### ISSUE TITLE
Unify haskell-indentation defcustoms into one variable

### ISSUE THREAD

**Issue by gracjan** [haskell/haskell-mode#952]
All of those should be replaced by a single defcustom:

    '(haskell-indentation-ifte-offset 4)
    '(haskell-indentation-layout-offset 4)
    '(haskell-indentation-left-offset 4)
    '(haskell-indentation-starter-offset 4)
    '(haskell-indentation-where-post-offset 4)
    '(haskell-indentation-where-pre-offset 4)

That defcustom should inherit value from `c-basic-offset`.

**Comment by emmanueltouzery** [comment]
I think there are people who have a `do` offset at 4 and a `where` offset at 2. I would not merge those values.

**Comment by Shimuuar** [comment]
Indeed. For example Johan Tibell: https://github.com/tibbe/haskell-style-guide/blob/master/haskell-style.md

**Comment by gracjan** [comment]
Tibell style has only where as exception. This is way less than 6 special
undocumented cases that we have now.

W dniu czwartek, 29 października 2015 Aleksey Khudyakov <
notifications@github.com> napisał(a):

**Comment by bgamari** [comment]
It's actually unclear to how one is supposed to replicate @tibbe's style with `haskell-indentation` currently. I currently use,
```lisp
 '(haskell-indentation-ifte-offset 4)
 '(haskell-indentation-layout-offset 4)
 '(haskell-indentation-left-offset 4)
```
as recommended in [`haskell-style.el`](https://github.com/tibbe/haskell-style-guide/blob/master/haskell-style.el) and yet bindings in `where` clauses are indented two spaces too far,
```hs
hello hi = do
    doThings aBinding
  where
      aBinding = hi
```
What am I missing here?

**Comment by gracjan** [comment]
@bgamari: looks like a bug to me.

**Comment by bgamari** [comment]
Note that the indentation issue that I mention above came up again in #1042.
